lib/metasploit/framework/login_scanner/mssql.rb

```diff
@@ -68,6 +68,7 @@ def attempt_login(credential)
           begin
             if mssql_login(credential.public, credential.private, '', credential.realm)
               result_options[:status] = Metasploit::Model::Login::Status::SUCCESSFUL
+              disconnect
```

Comment:
I think we should put this in an ensure block, similar to 

https://github.com/rapid7/metasploit-framework/blob/ad4b47f6ac6b3dfb9b55841c24d6bfb930e800d5/lib/metasploit/framework/login_scanner/smb.rb#L164-L165

Reply:
cc @sjanusz-r7 I think we'll need a similar pattern for your postgres stuff here too - since it doesn't use an ensure block 🤔 

https://github.com/rapid7/metasploit-framework/blob/ad4b47f6ac6b3dfb9b55841c24d6bfb930e800d5/lib/metasploit/framework/login_scanner/postgres.rb#L72-L77